youtube_dl/extractor/openload.py

```diff
@@ -58,68 +62,88 @@ class OpenloadIE(InfoExtractor):
         'only_matching': True,
     }]
 
+    _PHANTOMJS_SCRIPT = r'''
+        phantom.onError = function(msg, trace) {
+          var msgStack = ['PHANTOM ERROR: ' + msg];
+          if(trace && trace.length) {
+            msgStack.push('TRACE:');
+            trace.forEach(function(t) {
+              msgStack.push(' -> ' + (t.file || t.sourceURL) + ': ' + t.line
+                + (t.function ? ' (in function ' + t.function +')' : ''));
+            });
+          }
+          console.error(msgStack.join('\n'));
+          phantom.exit(1);
+        };
+        var page = require('webpage').create();
+        page.settings.resourceTimeout = 10000;
+        page.onInitialized = function() {
+          page.evaluate(function() {
+            delete window._phantom;
+            delete window.callPhantom;
+          });
+        };
+        page.open('https://openload.co/embed/%s/', function(status) {
+          var info = page.evaluate(function() {
+            return {
+              decoded_id: document.getElementById('streamurl').innerHTML,
+              title: document.querySelector('meta[name="og:title"],'
+                + 'meta[name=description]').content
+            };
+          });
+          console.log(info.decoded_id + ' ' + info.title);
+          phantom.exit();
+        });'''
+
     @staticmethod
     def _extract_urls(webpage):
         return re.findall(
             r'<iframe[^>]+src=["\']((?:https?://)?(?:openload\.(?:co|io)|oload\.tv)/embed/[a-zA-Z0-9-_]+)',
             webpage)
 
     def _real_extract(self, url):
+        exe = check_executable('phantomjs', ['-v'])
+        if not exe:
```

Comment:
I'm not sure where to put that code. Access to cookie jar would be nice but I don't know where it is.